sys/net/application_layer/dhcpv6/client.c

```diff
@@ -566,6 +648,52 @@ static void _parse_advertise(uint8_t *adv, size_t len)
                     }
                 }
                 break;
+#if IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_NA)
+            case DHCPV6_OPT_IA_NA:
+                for (unsigned i = 0; i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX; i++) {
```

Comment:
In https://github.com/RIOT-OS/RIOT/pull/16658 I just did

```suggestion
            case DHCPV6_OPT_IA_NA:
                for (unsigned i = 0;
                    IS_USED(MODULE_GNRC_DHCPV6_CLIENT_IA_NA) &&
                    (i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX);
                    i++) {
```

This way you don't have the `#if` and prevent an extra level of indentation. Plus, the option is still ignored when IA_NA is not compiled in.